src/node_buffer.cc

```diff
@@ -1213,6 +1213,10 @@ void Initialize(Local<Object> target,
     // TODO(thangktran): drop this check when V8 is pumped to 8.0 .
     if (!array_buffer->IsExternal())
       array_buffer->Externalize(array_buffer->GetBackingStore());
+    array_buffer->SetPrivate(
+        env->context(),
+        env->arraybuffer_untransferable_private_symbol(),
+        True(env->isolate())).Check();
```

Comment:
@thangktran - could you please explain this in a one liner, in the context of https://github.com/nodejs/node/issues/31061#issuecomment-568355014 ? as to how `SetPrivate` call is going to help here? Is it that with this change, the array buffer is not going to be registered in the BackingStore?

Reply:
also just wondering whether  similar thing to do be done here as well? https://github.com/nodejs/node/blob/5707ed21a253b2ed1e2f5944f622c20092b866d6/src/node_buffer.cc#L492-L493

Reply:
> @thangktran - could you please explain this in a one liner, in the context of [#31061 (comment)](https://github.com/nodejs/node/issues/31061#issuecomment-568355014) ? as to how `SetPrivate` call is going to help here?

Fwiw, I don’t see any way in which this might affect #31061.

> Is it that with this change, the array buffer is not going to be registered in the BackingStore?

No. This PR only ensures that the ArrayBuffer cannot be transferred through a `MessagePort`.

> also just wondering whether similar thing to do be done here as well?

I don’t see any reason to do that.

Reply:
> > @thangktran - could you please explain this in a one liner, in the context of [#31061 (comment)](https://github.com/nodejs/node/issues/31061#issuecomment-568355014) ? as to how `SetPrivate` call is going to help here?
> 
> Fwiw, I don’t see any way in which this might affect #31061.
> 

For the record, I ran some stress tests (https://github.com/nodejs/node/issues/31061#issuecomment-568484779) against this PR and can confirm it doesn’t fix https://github.com/nodejs/node/issues/31061.

Reply:
(Stress tests for https://github.com/nodejs/node/issues/31061 do seem to indicate it was caused by https://github.com/nodejs/node/commit/4f523c2c1a1c4cef33a1f925cb9102d5b8a51dab but I’m assuming we don’t want to revert that.)

Reply:
Yeah, I’ve added an explanation for the flakiness in #31061. It’s really not related to this.